ui/lib/sync/addon/utils/api-transforms.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { findDestination } from 'core/helpers/sync-destinations';
+
+import type { DestinationType, ListDestination } from 'vault/sync';
+import type { SystemListSyncDestinationsResponse } from '@hashicorp/vault-client-typescript';
+
+export const listDestinationsTransform = (
+  response: SystemListSyncDestinationsResponse,
+  nameFilter?: string,
+  typeFilter?: string
+) => {
+  const { keyInfo } = response;
+  const destinations: ListDestination[] = [];
+  // loop through each destination type (keys in key_info)
```

Comment:
The loop is given, but it's helpful to know what you're building with it
```suggestion
  // Build full destination objects from keyInfo
```

Reply:
I'm not sure that your suggestion is accurate since this isn't a destination but a combination of static destination information along with the `keyInfo` which is partial destination information. The type for it is `ListDestination` so I will add this suggestion to the batch commit and update it to `build ListDestination objects from keyInfo`